disnake/state.py

```diff
@@ -1880,6 +1881,28 @@ def parse_auto_moderation_action_execution(
         event = AutoModActionExecution(data=data, guild=guild)
         self.dispatch("automod_action_execution", event)
 
+    def parse_guild_audit_log_entry_create(self, data: gateway.AuditLogEntryCreate) -> None:
+        guild = self._get_guild(int(data["guild_id"]))
+        if guild is None:
+            _log.debug(
+                "GUILD_AUDIT_LOG_ENTRY_CREATE referencing unknown guild ID: %s. Discarding.",
+                data["guild_id"],
+            )
+            return
+
+        entry = AuditLogEntry(
+            data=data,
+            guild=guild,
+            application_commands={},
+            automod_rules={},
+            guild_scheduled_events=guild._scheduled_events,
+            integrations={},
+            threads=guild._threads,
+            users=dict(self._users),
```

Comment:
this works great, but the `dict()` call can become fairly expensive with large caches

Since `WeakValueDictionary` is a `MutableMapping`, `AuditLogEntry.__init__` should accept `Mapping`s instead of `Dict`s to make this work without casting.

Reply:
dc3dc27819627d92d5a3f04ae0d82f1f556d042c :)